services/src/functional-tests/src/main/groovy/org/openkilda/functionaltests/extension/fixture/CleanupVerifierExtension.groovy

```diff
@@ -32,23 +33,25 @@ class CleanupVerifierExtension extends ContextAwareGlobalExtension {
         if (!enabled) {
             return
         }
-        spec.addListener(new AbstractRunListener() {
-            @Override
-            void afterSpec(SpecInfo runningSpec) {
-                log.info("Running cleanup verifier for '$runningSpec.name'")
-                assert northbound.getAllFlows().empty
-                northbound.getAllSwitches().each {
-                    def validation = northbound.validateSwitch(it.switchId)
-                    validation.verifyRuleSectionsAreEmpty()
-                    validation.verifyMeterSectionsAreEmpty()
+        spec.features.each {
+            it.addInterceptor(new IMethodInterceptor() {
+                @Override
+                void intercept(IMethodInvocation invocation) throws Throwable {
+                    invocation.proceed()
+                    log.info("Running cleanup verifier for '$invocation.feature.name'")
+                    assert northbound.getAllFlows().empty
+                    northbound.getAllSwitches().each {
+                        def validation = northbound.validateSwitch(it.switchId)
+                        validation.verifyRuleSectionsAreEmpty()
+                        validation.verifyMeterSectionsAreEmpty()
```

Comment:
I guess we should check that meter is supported by a switch (`.description.contains("OF_12")` )

Reply:
`verifyMeterSectionsAreEmpty` inside verifies that the 'meters' section is present in the validation response. So it's already safe